packages/webrtc-core/src/media/local-track.ts

```diff
@@ -0,0 +1,107 @@
+/* eslint-disable no-underscore-dangle */
+import { EventMap } from '../event-emitter';
+import { logger } from '../util/logger';
+import { Events, Track, TrackState } from './track';
+
+export type TrackEndEvent = {
+  trackState: TrackState;
+};
+
+export type TrackMuteEvent = {
+  trackState: TrackState;
+};
+
+export type TrackPublishEvent = {
+  isPublished: boolean;
+  trackState: TrackState;
+};
+
+export interface LocalTrackEvents extends EventMap {
+  [Events.PublishedStateUpdate]: (event: TrackPublishEvent) => void;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type TrackEffect = any;
+// TBD: Fix this once types are published separately
+// export type TrackEffect = BaseMicrophoneEffect | BaseCameraEffect;
```

Comment:
do we need to have the effects published separately before merging this?

Reply:
Its already published seperately but in the webrtc core lib the only need is the types , 

@andiwils  any suggestion on how to just import the types in the webrtc core we dont want to include the code as the developer will import the effects lib and pass to us